_source/_posts/2021-07-14-flask-scim-server.md

```diff
@@ -0,0 +1,519 @@
+
+#### Assign a Group
+Under the **Assignments** tab, click **Assign** -> **Assign to Groups**. I assigned group **Droids**. Note that this acts similar to the above assigning of individual users. Okta will iterate through the group membership and create them in the external SCIM server - but the Group itself won't be made. That is done in **Push Groups** which we will handle later. Here is a look at the requests and responses from Okta to the SCIM server:
+
+ - *Okta Request:*
+```
+GET /scim/v2/Users?filter=userName%20eq%20%22c-3po%40iamciam.dev%22&startIndex=1&count=100
+```
+ - *SCIM Server Response:*
+```
+200 OK
+```
+```
+{
+    "Resources": [],
+    "itemsPerPage": 0,
+    "schemas": [
+        "urn:ietf:params:scim:api:messages:2.0:ListResponse"
+    ],
+    "startIndex": 1,
+    "totalResults": 0
+}
+```
+ - *Okta Request:*
+```
+POST /scim/v2/Users
+```
+```
+{
+    "schemas": [
+        "urn:ietf:params:scim:schemas:core:2.0:User"
+    ],
+    "userName": "c-3po@iamciam.dev",
+    "name": {
+        "givenName": "C",
+        "familyName": "3PO"
+    },
+    "emails": [
+        {
+            "primary": true,
+            "value": "c-3po@iamciam.dev",
+            "type": "work"
+        }
+    ],
+    "displayName": "C 3PO",
+    "locale": "en-US",
+    "externalId": "00usemeo53HWWPYy14x6",
+    "groups": [],
+    "password": "wA&&cprB",
+    "active": true
+}
+```
+ - *SCIM Server Response:*
+```
+201 CREATED
+```
+```
+{
+    "active": true,
+    "displayName": "C 3PO",
+    "emails": [
+        {
+            "primary": true,
+            "type": "work",
+            "value": "c-3po@iamciam.dev"
+        }
+    ],
+    "externalId": "00usemeo53HWWPYy14x6",
+    "groups": [],
+    "id": "478c4cf4-1aa3-41a1-93da-4c154c5955e0",
+    "locale": "en-US",
+    "meta": {
+        "resourceType": "User"
+    },
+    "name": {
+        "familyName": "3PO",
+        "givenName": "C",
+        "middleName": null
+    },
+    "schemas": [
+        "urn:ietf:params:scim:schemas:core:2.0:User"
+    ],
+    "userName": "c-3po@iamciam.dev"
+}
+```
+
+- There are 3 members in my **Droids** group: C-3P0, R2-D2 and BB-8. The above req/resp chain repeats for the other 2 users. Now when I run `select * from users;` in the scim database I can see the assigned users:
+
+{% img blog/flask-scim-server/scim_11.png alt:"Database Assigned Users Image" width:"800" %}{: .center-image }
+
+#### Push Groups
+
+As mentioned above, assigning a group in Okta only adds the users to my SCIM server, not the Group itself. In order to add the Group, tab over to **Push Groups** in your Okta app integration. Click **Push Groups** -> **Find groups by name** and select the group you assigned above. **Make sure Push group memberships immediately** is checked and click **Save**. Two calls should be made by Okta - one creating the group itself and another updating the group membership. Here are the requests and responses for my Droids group push:
```

Comment:
Double check the formatting of the this:

```txt
**Make sure Push group memberships immediately**
```

Ideally bold the name of the checkbox/button, or to really call it out use a "NOTE" section or a quote call out, similar to what you have above